engine/src/main/java/com/arcadedb/query/sql/executor/CartesianProductStep.java

```diff
@@ -177,7 +177,7 @@ public void addSubPlan(InternalExecutionPlan subPlan) {
 
   @Override
   public String prettyPrint(int depth, int indent) {
-    String result = "";
```

Comment:
This parts are tests or parts that it's not a problem if they are optimized. I prefer keeping the string concatenation that is more readable. I've merged other commits where it might affect performance.